pkg/kubectl/describe.go

```diff
@@ -477,8 +477,9 @@ func (d *ServiceDescriber) Describe(namespace, name string) (string, error) {
 
 	endpoints, _ := d.Endpoints(namespace).Get(name)
 	events, _ := d.Events(namespace).Search(service)
+	pods, _ := d.Pods(namespace).List(labels.Set(service.Spec.Selector).AsSelector(), fields.Everything())
 
-	return describeService(service, endpoints, events)
+	return describeService(service, endpoints, events, pods.Items)
```

Comment:
Endpoints has the additional advantages that dead and not-ready pods are filtered out, and that it would work for manually populated Endpoints (for services w/o selectors).